### PR TITLE
Add NonNullApi annotation to all internal packages via package info

### PR DESCRIPTION
**Pull request by alllex**
TBD

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test this

File: build-logic/uber-plugins/src/main/kotlin/gradlebuild.java-library.gradle.kts [hunk 1]
```diff
@@ -33,3 +33,9 @@ plugins {
     id("gradlebuild.ci-reporting") // CI: Prepare reports to be uploaded to TeamCity
     id("gradlebuild.configure-ci-artifacts") // CI: Prepare reports to be uploaded to TeamCity
 }
+
+dependencies {
+    if (project.name != "base-annotations") {
+        compileOnly(project(":base-annotations"))
+    }
+}
```
Comment (octylFractal) [inline build-logic/uber-plugins/src/main/kotlin/gradlebuild.java-library.gradle.kts:41]: I think this is too broad -- I'm not sure projects like `performance` need this dependency. It should probably be added to each `build.gradle.kts` as needed.

**Comment by octylFractal** [comment]
As a follow-up we should remove the annotation where it isn't needed any more, i.e. on specific methods or classes in packages that now have the annotation.

**Comment by alllex** [comment]
Just adding the `package-info.java` files with the annotation was not enough. The stricter nullability was taken into account by Kotlin compilation, which lead to compilation failures.

I would suggest that we annotate those affected packages properly in separate PRs, making it, so this PR would **only** add the info files.

**Pull request closed without merging**